data_types/softlayer_dns_domain.go

```diff
@@ -0,0 +1,20 @@
+package data_types
+
+type SoftLayer_Dns_Domain_Template struct {
+	Name            string                                 `json:"name"`
+	ResourceRecords []SoftLayer_Dns_Domain_Resource_Record `json:"resourceRecords"`
+}
+
+type SoftLayer_Dns_Domain_Template_Parameters struct {
+	Parameters []SoftLayer_Dns_Domain_Template `json:"parameters"`
+}
+
+type SoftLayer_Dns_Domain struct {
+	Id         int    `json:"id"`
```

Comment:
We plan to start using `uint32` for IDs and other ID-related fields. This will likely be a sweeping change for v2 but might be good to be ahead of the curve. Can you try that with your ID fields if you have time?